src/AstAnalyser.js

```diff
@@ -14,10 +15,25 @@ import { JsSourceParser } from "./JsSourceParser.js";
 export class AstAnalyser {
   /**
    * @constructor
-   * @param { SourceParser } [parser]
+   * @param options
    */
-  constructor(parser = new JsSourceParser()) {
-    this.parser = parser;
+  constructor(options = {}) {
+    if (options.customParser !== undefined) {
+      assert(options.customParser instanceof JsSourceParser || typeof options.customParser === "object",
+        `customParser must be an instance of JsSourceParser or an object`);
```

Comment:
```suggestion
        "customParser must be an instance of JsSourceParser or an object");
```

Reply:
according to this com https://github.com/NodeSecure/js-x-ray/pull/250#discussion_r1526343856 this part has been removed